lib/fluent/plugin/out_derive.rb

```diff
@@ -170,6 +171,19 @@ def calc_rate(tag, key, cur_value, prev_value, cur_time, prev_time, adjustment =
     end
   end
 
+  def _calc_rate(cur_value, prev_value)
+    rate = cur_value - prev_value
+    if @counter_mode
+      if rate < 0
+        rate += (1 << 32 - 1)
+      end
+      if rate < 0
+        rate += (1 << 64 - 1 << 32 )
```

Comment:
This also returns unwanted value. `-` operator is higher priority than `<<`.

How about `2**32-1` and `2**64 - 2**32`? `**` is higher than `-`.